### DISCUSSION TITLE
Why does Neovim touch file "4913" when writing?

### DISCUSSION THREAD

**Discussion by camoz** [neovim/neovim#30613] [General]
Hi,

I noticed that when I modify and write a file using the default nvim configuration, i.e. via `nvim --clean test.txt`, and run inotifywait alongside, there is a temporary file created and deleted before writing the actual file. E.g. the command `inotifywait -m -e modify -e attrib -e close_write -e move -e create -e delete .` outputs:

```
./ CREATE 4913
./ ATTRIB 4913
./ CLOSE_WRITE,CLOSE 4913
./ DELETE 4913
./ MOVED_FROM test.txt
./ MOVED_TO test.txt~
./ CREATE test.txt
./ MODIFY test.txt
./ CLOSE_WRITE,CLOSE test.txt
./ ATTRIB test.txt
./ DELETE test.txt~
```

I mean the file `4913`, not the backup file `test.txt~`.

Any idea where this comes from?

**Comment by justinmk** [comment]
this is a classic question. it comes from here: https://github.com/neovim/neovim/blob/d3b4772ddcd4c890dc5bc38e1f1b36a08aed0c04/src/nvim/bufwrite.c#L740-L752

see also https://groups.google.com/g/vim_dev/c/sppdpElxY44

**Reply by camoz** [reply]
Ahh I see, thanks!

(I did google it but did not include "4913" in the search because I thought it's a random number... so I didn't find anything.)
